cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx

```diff
@@ -86,6 +91,28 @@ const TemplateStore = () => {
       });
   };
 
+  const checkTemplateReference = (repo: any, path: any, templateName: any) => {
+    setLoadingTemplateName(templateName);
+    axios
+      .get(`/api/templates?repo=${repo}&path=${path}`)
```

Comment:
You will need to pass the version here as well. 
```suggestion
      .get(`/api/templates?repo=${repo}&path=${path}&commit=${version}`)
```

Reply:
I was thinking about this but I didn't do it at the first place as the version was not required during the template creation.

Reply:
For templates stored on git, cyclops defaults to the default branch, but it fails for OCI and helm repo charts